middleware/app/Http/Controllers/Api/V1/ConsumerPlatformController.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers\Api\V1;
+
+use App\Actions\Fortify\UpdateUserPassword;
+use App\Models\ConsumerPlatform;
+use ConsumerPlatformServices;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Api\V1\Controller;
+use App\Http\Requests\StoreConsumerPlatformRequest;
+use App\Http\Requests\UpdateConsumerPlatformRequest;
+use App\Http\Resources\ConsumerPlatformResource;
+
+class ConsumerPlatformController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     * GET: /api/consumer_platforms/
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function index(): \Illuminate\Http\JsonResponse
+    {
+        return response()->json(json_encode(ConsumerPlatform::all()));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     * POST /api/consumer_platforms/
+     * @param StoreConsumerPlatformRequest $request
+     * @return ConsumerPlatformResource
+     */
+    public function store(StoreConsumerPlatformRequest $request): ConsumerPlatformResource
+    {
+        try {
+            $validated = $request->validated();
+            $cp = ConsumerPlatform::create($validated);
+        } catch (\Exception) {
+            return response()->json(['error' => 'invalid request'], 400);
+        }
+        return new ConsumerPlatformResource($cp);
```

Comment:
Is there a reason to use `new` instead of `ConsumerPlatformResource::make($cp);`?

Reply:
As discussed, returning `new X` is :+1: but using `new X` then trying to update it and using `save()` is :-1: 

So for update functions, `make` + `save` is the way to go :+1: